core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -209,11 +224,12 @@ public void testSubscribeSubscribeEmptyPollFails() {
 
     @Test
     public void testSubscriptionAndPoll() {
+        KafkaShareConsumer<byte[], byte[]> shareConsumer = createShareConsumer(new ByteArrayDeserializer(), new ByteArrayDeserializer(), "group1");
+        shareConsumer.subscribe(Collections.singleton(tp.topic()));
+        alterShareAutoOffsetReset("group1", "earliest");
```

Comment:
Do we have test without specifying the offset i.e. when default offset is pulled?

Reply:
Why to change all tests rather add a test where we do specific updates to group offset using admin client?

Reply:
Thanks for the review. The issue here is that the default value of share.auto.offset.reset config is "latest". Also the share partition is initialized during the first share fetch request to that topic partition for a groupId, which means during share partition initialization, if the default value is returned by persister, then by default the value of start offset will be set to the latest offset for that topic partition, thereby discarding all the previously produced messages to that topic partition. And, in almost every test, the share fetch requests (issues by shareConsumer.poll) are issued after some messages have been produced to that that topic. So, in order to not discard those previous messages and thus test the specific functionality being tested in the tests, I have changed the value of the config to "earliest", which would simply start fetching from the first record and doesn't care for the order in which messages are produced and consumed.

Reply:
is there a way by which we can change the default only for our tests? That would be better, I guess

Reply:
Thanks @chirag-wadhwa5 , I think then it's fine to add this additional line.

Reply:
Though not sure why you have to move the order of the calls? Previously we were producing and then consuming. Now we start consuming and then produce. Why?

Isn't produce and then consumer from earliest will give same results?

Reply:
Thanks for the review. This is actually a piece of code that I had in mind to revert. It was required earlier when I was taking a different approach to resolve the failing tests by making a share fetch request before producing any messages. But since changing the value of the config itself is a better solution, this is not required anymore. I have made the changes and will push it in the next commit. Thanks !